packages/plugins/typescript/typed-document-sdk/src/sdk-base.ts

```diff
@@ -85,15 +78,17 @@ type SDKSelectionKeysWithoutArguments<TSelection extends SDKSelection> = Exclude
 type SDKOperationType<TSelection extends SDKSelection, TResultType extends ResultType> = {
   // check whether field in in result type
   [TSelectionField in SDKSelectionKeysWithoutArguments<TSelection>]: TSelectionField extends keyof TResultType
-    ? null extends TResultType[TSelectionField]
+    ? TSelection[TSelectionField] extends boolean
+      ? TResultType[TSelectionField]
+      : null extends TResultType[TSelectionField]
       ? SDKOperationTypeInner<TSelection[TSelectionField], SDKNonNullable<TResultType[TSelectionField]>> | null
       : SDKOperationTypeInner<TSelection[TSelectionField], TResultType[TSelectionField]>
     : never;
 };
 
-export type SDKSelectionSet<TType extends Record<string, any> = any> = AtLeastOnePropertyOf<NoExtraProperties<TType>>;
+export type SDKSelectionSet<TType> = AtLeastOnePropertyOf<NoExtraProperties<TType>>;
 
-export type SDKUnionSelectionSet<TType extends Record<string, any> = any> = AtLeastOnePropertyOf<TType>;
+export type SDKUnionSelectionSet<TType extends Record<string, any> = any> = TType; // AtLeastOnePropertyOf<TType>;
```

Comment:
@n1ru4l is this expected? 👀

Reply:
already fixed locally, will push it soon!